src/script/components/Modals/PrimaryModal/PrimaryModal.tsx

```diff
@@ -0,0 +1,222 @@
+                  <input
+                    className="modal__input"
+                    type="password"
+                    value={passwordValue}
+                    placeholder={inputPlaceholder ?? ''}
+                  />
+                </form>
+              )}
+              {hasInput && (
+                <form onSubmit={() => doAction(confirm, !!closeOnConfirm)}>
+                  <input
+                    maxLength={64}
+                    className="modal__input"
+                    value={inputValue}
+                    placeholder={inputPlaceholder ?? ''}
+                  />
+                </form>
+              )}
+              {errorMessage && <div className="modal__input__error">{errorMessage}</div>}
+              {hasOption && (
+                <div className="modal-option">
+                  <div className="checkbox accent-text">
+                    <input
+                      className="modal-checkbox"
+                      type="checkbox"
+                      id="clear-data-checkbox"
+                      checked={optionChecked}
+                      data-uie-name="modal-option-checkbox"
+                    />
+                    <label className="label-xs" htmlFor="clear-data-checkbox">
+                      <span className="modal-option-text text-background">{checkboxLabel}</span>
+                    </label>
+                  </div>
+                </div>
+              )}
+              <div className={cx('modal__buttons', {'modal__buttons--column': hasMultipleSecondary})}>
+                {secondaryActions
+                  .filter((action): action is Action => action !== null)
+                  .map(action => (
+                    <button
+                      key={`${action?.text}-${action?.uieName}`}
+                      type="button"
+                      onClick={() => action?.action && doAction(action?.action, true, true)}
+                      data-uie-name={action?.uieName}
+                      className={cx('modal__button modal__button--secondary', {
+                        'modal__button--full': hasMultipleSecondary,
+                      })}
+                    >
+                      {action?.text}
+                    </button>
+                  ))}
+                {primaryAction?.text && (
+                  <button
+                    type="button"
+                    onClick={() => doAction(confirm, !!closeOnConfirm)}
+                    disabled={!actionEnabled}
+                    className={cx('modal__button modal__button--primary', {
+                      'modal__button--full': hasMultipleSecondary,
+                    })}
+                    data-uie-name="do-action"
+                  >
+                    {primaryAction.text}
+                  </button>
+                )}
+              </div>
+            </div>
+          </>
+        )}
+      </ModalComponent>
+    </div>
+  );
+};
+
+const PrimaryModal = {
+  add: addNewModalToQueue,
```

Comment:
```suggestion
  show: addNewModalToQueue,
```

I think `show` makes a little more sense when we talk about modals. WDYT?